scenes/game/local_game.gd

```diff
@@ -1842,8 +1851,8 @@ class Player:
 			parent._append_log_full(Enums.LogType.LogType_CardInfo, self, "reveals the top card of their deck to the opponent.")
 		else:
 			parent._append_log_full(Enums.LogType.LogType_CardInfo, self, "reveals the top card of their deck: %s." % card_name)
+			public_topdeck_id = deck[0].id
```

Comment:
I think this should still happen in both cases?
The public state should be the same on both clients.

Reply:
The issue I ran into was that Anji could peek at your topdeck, and if you have a copy of the same card in your hand you can see the icon there

Reply:
Off the top of my head I think it's always just either only you or only the opponent? unsure if anything similar happens for "look at the top card of your deck" type effects